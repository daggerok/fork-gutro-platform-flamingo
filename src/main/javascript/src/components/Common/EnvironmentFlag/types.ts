export interface Environments {
  italy: string;
  spain: string;
  malta: string;
  dev: string;
}
