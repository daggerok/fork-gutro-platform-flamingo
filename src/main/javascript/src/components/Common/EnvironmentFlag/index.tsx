import React, { useEffect, useState } from 'react';
import { ApiResult } from '../../../types';
import { fetchEnvironment } from '../../../api/environment';
import { Environments } from './types';
import Flag from './../Flag';

const EnvironmentFlag: React.FC = () => {
  const [environment, setEnvironment] = useState<string>('');
  const [country, setCountry] = useState<string>('');

  const FLAG_MAP: Environments = {
    italy: 'IT',
    spain: 'ES',
    malta: 'MT',
    dev: '',
  };

  useEffect(() => {
    if (!environment) { return; }
    setCountry(FLAG_MAP[environment as keyof Environments]);
  }, [environment]);

  const getEnvironment = async (): Promise<ApiResult> => {
    return fetchEnvironment()
      .then((response) => {
        return response;
      })
      .catch((error) => {
        throw new Error(error); 
      });
  };

  useEffect(() => {
    const initEnvironment = async (): Promise<void> => {
      const { data } = await getEnvironment();
      setEnvironment(data.environment);
    };
    initEnvironment().catch(e => console.error(e));
  }, []);

  return (
    <Flag 
      country={country}  
    />
  );
};

export default EnvironmentFlag;
