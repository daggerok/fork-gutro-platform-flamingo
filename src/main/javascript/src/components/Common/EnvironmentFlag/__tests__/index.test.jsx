import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import EnvironmentFlag from '..';

jest.mock('./../../Flag', () => 'mock-flag');

let component = null;

describe('<EnvironmentFlag />', () => {

  beforeAll(() => {
    act(() => {
      component = mount(<EnvironmentFlag />)
    });
  });

  it('matches snapshot', () => {
    expect(component.html()).toMatchSnapshot()
  });

  afterAll(() => {
    component.remove();
    component = null;
  });

}); 


   