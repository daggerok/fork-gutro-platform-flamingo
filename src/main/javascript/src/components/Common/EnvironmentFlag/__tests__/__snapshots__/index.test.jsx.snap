// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnvironmentFlag /> matches snapshot 1`] = `"<mock-flag country=\\"\\"></mock-flag>"`;
